public/pages/main/Main.tsx

```diff
@@ -15,50 +15,106 @@
 
 import { Switch, Route, RouteComponentProps, Redirect } from 'react-router-dom';
 import React from 'react';
+import { AppState } from '../../redux/reducers';
 import { CreateDetector } from '../createDetector';
-
 import { DetectorList } from '../DetectorsList';
 import { ListRouterParams } from '../DetectorsList/List/List';
-import { PreviewDetector } from '../PreviewDetector/containers/PreviewDetector';
-import { Dashboard } from '../Dashboard/Container/Dashboard';
-import { APP_PATH } from '../../utils/constants';
+// @ts-ignore
+import { EuiSideNav, EuiPage, EuiPageBody, EuiPageSideBar } from '@elastic/eui';
+import { useSelector } from 'react-redux';
+
+enum Navigation {
+  AnomalyDetection = 'Anomaly detection',
+  Dashboard = 'Dashboard',
+  Detectors = 'Detectors',
+}
 
-interface MainProps {}
+enum Pathname {
+  Dashboard = '/dashboard',
+  Detectors = '/detectors',
+}
+
+interface MainProps extends RouteComponentProps {}
+
+export function Main(props: MainProps) {
+  const hideSideNavBar = useSelector(
+    (state: AppState) => state.adApp.hideSideNavBar
+  );
```

Comment:
@ylwu-amzn  Do you think we would need this state globally ? I think it can be just use local state v/s global state in redux. Do you need this value of state outside of this component ? If not let's just `useState` hooks from react and remove all Redux layer if not really needed.

Reply:
Thanks, we need to use this state for multiple components/pages. For the home page, we should show side nav bar, but for detector detail, AD result page, edit detector/feature pages, we should hide the side nav bar. So I think we need a global state in redux.